diff/diff.go

```diff
@@ -281,8 +282,12 @@ func (sc *Syncer) eventCompleted() {
 func (sc *Syncer) wait() {
 	time.Sleep(time.Duration(sc.StageDelaySec) * time.Second)
 	for atomic.LoadInt32(&sc.inFlightOps) != 0 {
-		// TODO hack?
-		time.Sleep(1 * time.Millisecond)
+		select {
+		case <-sc.stopChan:
+			return
```

Comment:
Thinking about this - could it ever happen that `stopChan` is closed while we are still waiting for legitimate operations, e.g. if we are syncing up 100s of entities and it is taking a while? The `close(sc.stopChan)` statement seems pretty standalone.

Reply:
This should still be okay because returning from here will make the producer stop. The consumers will still continue to run to completion (either successfully or error out).